builder/fscache/fscache.go

```diff
@@ -607,3 +616,40 @@ func (s sortableCacheSources) Less(i, j int) bool {
 func (s sortableCacheSources) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
+
+func newTarsumHash(stat *fsutil.Stat) (hash.Hash, error) {
```

Comment:
this code was previously in https://github.com/moby/moby/pull/34943/files#diff-3b65b3cb4440e2f8d672b27e896f75a8L314